srml/contract/src/lib.rs

```diff
@@ -14,41 +14,69 @@
 // You should have received a copy of the GNU General Public License
 // along with Substrate. If not, see <http://www.gnu.org/licenses/>.
 
-//! Smart-contract module for runtime; Allows deployment and execution of smart-contracts
-//! expressed in WebAssembly.
+//! # Contract Module
+//! 
+//! The contract module provides functionality for the runtime to deploy and execute smart-contracts expressed in WebAssembly.
+//! To add this functionality, you need to add the contract module to `construct_runtime!` in the runtime module.
+//! The supported dispatchable functions are documented as part of the [`Call`] enum.
+//! 
+//! ## Overview
+//! 
+//! This module extends accounts with the ability to create smart-contracts and make calls to the contracts.
+//! Each smart-contract is associated with an account, which handles the code and storage. When a smart-contract is called, its associated code gets executed.
+//! This code can alter the storage entries of the associated account, create new smart-contracts, or call other smart-contracts.
+//! 
+//! Senders must specify a gas limit with every call, as all instructions invoked by the smart-contract require gas.
+//! Unused gas is refunded after the call, regardless of the execution outcome. 
+//! 
+//! **NOTE:** If the gas limit is reached, then all calls and state changes (including balance transfers) are only reverted at the current call's contract level.
+//! For example, if contract A calls B and B runs out of gas mid-call, then all of B's calls are reverted.
+//! Assuming correct error handling by contract A, A's other calls and state changes still persist. 
+//! 
+//! **NOTE:** Call failures are also not always cascading. For example, if contract A calls contract B and B
+//! fails, A can decide how to handle that failure, either proceeding or reverting A's changes.
+//! 
+//! Finally, when the `balances` module determines an account is dead (i.e. account balance fell below the
+//! existential deposit), it reaps the account. This will delete the associated code and storage of the account.
+//! 
+//! ## Interface
+//! 
+//! ### Dispatchable functions ([`Call`])
```

Comment:
yes that doc is still in progress. Ill just remove and add back in once its ready

Reply:
I actually was hoping to define the reference because there is another reference to [`Call`] which is still undefined : )

<img width="636" alt="Screenshot 2019-03-26 at 12 19 54" src="https://user-images.githubusercontent.com/2205845/54993165-862eb080-4fc1-11e9-9be0-1fc548b8e0a3.png">

Reply:
oop already in progress by @shawntabrizi : https://github.com/paritytech/substrate-developer-hub/issues/12

I can also take a look, or give some PR to that once ready :)

Reply:
Hmmm I am not sure if I understand what is this issue about and what it has to do with this case. 

Anyway, if you take a look at [the current version of the docs](https://crates.parity.io/srml_contract/index.html), you can see [the `Call` enum](https://crates.parity.io/srml_contract/enum.Call.html). Each module has its [own definition of the `Call`](https://crates.parity.io/substrate_service/index.html?search=Call) enum. It looks like that we need to link it here.

Here is the definition of a link that should work:

```markdown
[`Call`]: ../enum.Call.html
```

it should be enough to just insert this line at the end of the doc.